doc/architectural/memory-primitives.md

```diff
@@ -91,6 +91,41 @@ point into another memory object (such as could happen when running on a real
 machine). To verify that pointers stay within the bounds of their pointees, the
 CBMC option `--pointer-overflow-check` can be used.
 
+### Malloc modelling
+
+CBMC ships a model of `malloc` that seeks to emulate the behaviour of the C
+standard library. By default, however, it will always succeed to allocate an
+object. This extends to the case of `malloc(0)`, which returns a valid pointer
+to an object. The size of that object is zero, implying that any attempt to read
+from or write to this object will result in an out-of-bounds access.
+
+The following command line options add emulation of failures:
+
+|Flag                    |  Check                                          |
+|------------------------|-------------------------------------------------|
+| `--malloc-fail-null`   |  in case malloc fails return NULL               |
```

Comment:
This needs more clarity. The wording gives the impression that the model might fail, but doesn't say that this extends the model with behavior that corresponds to a failure in a real implementation.

Reply:
Reworded.